### PR TITLE
move get_seed to randomness interface

### PR DESCRIPTION
**Pull request by rmudambi**

### CODE REVIEW

File: src/vivarium/framework/randomness.py [hunk 1]
```diff
@@ -627,10 +611,10 @@ def setup(self, builder):
 
         self.resources = builder.resources
         self._add_constraint = builder.lifecycle.add_constraint
-        builder.lifecycle.add_constraint(self.get_randomness_stream, allow_during=['setup'])
-        builder.lifecycle.add_constraint(self.register_simulants,
-                                         restrict_during=['initialization', 'setup', 'post_setup',
-                                                          'simulation_end', 'report'])
+        self._add_constraint(self.get_seed, restrict_during=['initialization'])
```
Comment (rmudambi) [inline src/vivarium/framework/randomness.py]: Can we extract these stage names as constants so we don't have strings everywhere?

File: src/vivarium/framework/randomness.py [hunk 2]
```diff
@@ -471,22 +471,6 @@ def get_draw(self, index: Index, additional_key: Any = None) -> pd.Series:
 
         return draw
 
-    def get_seed(self, additional_key: Any = None) -> int:
```
Comment (rmudambi) [inline src/vivarium/framework/randomness.py]: Is it better to deprecate rather than remove this?

Comment (collijk) [reply src/vivarium/framework/randomness.py]: In theory, yes.  We don't have a deprecation strategy (and is one of the reasons we're not at a v1 release).  So don't worry about it for now.

File: src/vivarium/framework/randomness.py (hunk 1 above)
Comment (collijk) [reply src/vivarium/framework/randomness.py]: Yes, but smartly.  Make a ticket in the next sprint backlog to survey string constants across vivarium and vph and then we'll come up with a strategy for handling categories of string constants.

File: src/vivarium/framework/randomness.py (hunk 2 above)
Comment (collijk) [reply src/vivarium/framework/randomness.py]: https://jira.ihme.washington.edu/browse/MIC-1614

File: src/vivarium/framework/randomness.py (hunk 1 above)
Comment (collijk) [reply src/vivarium/framework/randomness.py]: https://jira.ihme.washington.edu/browse/MIC-1615

**Pull request merged**